packages/cdk-assets/lib/private/docker.ts

```diff
@@ -158,18 +158,49 @@ export class Docker {
     const config = cdkCredentialsConfig();
     if (!config) { return false; }
 
-    this.configDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdkDockerConfig'));
+    // Condition configured for supporting testing, it is not expected the CDK_DOCKER_TMP_CONFIG_DIR to be set in production
+    if (process.env.CDK_DOCKER_TMP_CONFIG_DIR) {
```

Comment:
I advise against having separate code paths for test and production. It's very easy for someone in the future to make a mistake and change code in one path and not the other. Is there any reason you're not mocking the `fs.mkdtempSync` method?

Reply:
let me try that